apps/qubit/modules/digitalobject/actions/metadataComponent.class.php

```diff
@@ -36,6 +36,8 @@ public function execute($request)
     $this->storageServicePluginEnabled = (
       $this->context->getConfiguration()->isPluginEnabled('arStorageServicePlugin')
       && arStorageServiceUtils::getAipDownloadEnabled()
+      && $this->user->checkModuleActionAccess('arStorageService', 'download')
+      && $this->user->checkModuleActionAccess('arStorageService', 'extractFile')
```

Comment:
Is it necessary to check for both "download" and "extractFile" permissions? I would expect checking for 'download' authorization to be sufficient.

Reply:
This is because there are two plugin actions to protect and they _could_ be configured independently in the security.yml file (This code is using the security.yml file to determine user access): 

https://github.com/artefactual/atom/blob/qa/2.x/plugins/arStorageServicePlugin/modules/arStorageService/config/security.yml

I am checking both because the component being protected will display both download links.

Reply:
@sbreker I think it would be better to make the secuirty.yml checks specific to the relevant link then, rather then disabling both links when only one is authorized.

Reply:
@djjuhasz Good idea! See new commit - I have moved the checks into the Storage Service plugin WebUI component. This allows individual config of the two download links (see commit message).